### PR TITLE
support for TS.GET and TS.MGET

### PR DESCRIPTION
**Pull request by filipecosta90**
requires https://github.com/RedisTimeSeries/RedisTimeSeries/pull/321 to be merged upstream. 
- [add] support for TS.GET and TS.MGET  
- [add] Improved tests and documentation.
- [add] splitted client specific code, response parsing, and common methods
- [add] increased coverage

### CODE REVIEW

File: common.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+// Client is an interface to time series redis commands
+type Client struct {
+	Pool ConnPool
+	Name string
+}
+
+//
+//const TimeRangeMinimum = 0
+//const TimeRangeMaximum = math.MaxInt64
+//const TimeRangeFull = int64(-1)
```
Comment (itamarhaber) [inline common.go]: ?

File: client.go [hunk 2]
```diff
@@ -362,213 +108,152 @@ func (client *Client) AddWithRetention(key string, timestamp int64, value float6
...
+	reply, err = conn.Do("TS.MGET", args...)
+
+	if err != nil {
+		return
+	}
+	ranges, err = ParseRangesSingleDataPoint(reply)
+	return
+}
+
+// Info create a new time-series
```
Comment (itamarhaber) [inline client.go]: Wrong description

File: client.go [hunk 3]
```diff
@@ -362,213 +108,152 @@ func (client *Client) AddWithRetention(key string, timestamp int64, value float6
...
-			return nil, err
-		}
-		r := Range{name, labels, dataPoints}
-		ranges = append(ranges, r)
-	}
-	return ranges, nil
+// DeleteRule - delete a compaction rule
+// args:
+// SOURCE_KEY - key name for source time series
+// DEST_KEY - key name for destination time series
```
Comment (itamarhaber) [inline client.go]: Args style

File: client.go [hunk 4]
```diff
@@ -362,213 +108,152 @@ func (client *Client) AddWithRetention(key string, timestamp int64, value float6
...
-
-type Range struct {
-	Name       string
-	Labels     map[string]string
-	DataPoints []DataPoint
+// TS.CREATERULE create a compaction rule
+// SOURCE_KEY - key name for source time series
+// AGG_TYPE - AggregationType
+// BUCKET_SIZE_SEC - time bucket for aggregated compaction,
+// DEST_KEY - key name for destination time series
```
Comment (itamarhaber) [inline client.go]: Reformat comment to go-doc

File: common.go (hunk 1 above)
Comment (filipecosta90) [reply common.go]: For some reason I comment it out. Fixing on new commit :)

File: client.go (hunk 2 above)
Comment (filipecosta90) [reply client.go]: Fixing on new commit and revising all methods description

File: client.go (hunk 3 above)
Comment (filipecosta90) [reply client.go]: Fixing on new commit and revising all methods description

**Pull request merged**